File: code_samples/dart/file_operations_compression.dart
// File Operations: Compression
// AI/ML Training Sample

class Compression {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Compression();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/elixir/file_operations_compression.ex
# File Operations: Compression
# AI/ML Training Sample

defmodule Compression do
  defstruct data: ""
  
  def new(), do: %Compression{}
  
  def process(%Compression{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Compression{data: data}), do: data
  
  def validate(%Compression{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Compression.new()
updated = Compression.process(instance, "example")
IO.puts("Data: " <> Compression.get_data(updated))
IO.puts("Valid: " <> to_string(Compression.validate(updated)))
